backend/src/main/java/com/redhat/cloud/notifications/events/EventConsumer.java

```diff
@@ -122,50 +122,49 @@ public void init() {
                                     } else {
                                         /*
                                          * Step 4
-                                         * The message ID is new. We need to retrieve an EventType from the DB using the
-                                         * bundle/app/eventType triplet from the parsed Action.
+                                         * The message ID is new. Let's persist it. The current message will never
+                                         * be processed again as long as its ID stays in the DB.
                                          */
-                                        return appResources.getEventType(bundleName[0], appName[0], eventTypeName)
-                                                .onFailure(NoResultException.class).transform(e ->
-                                                        new NoResultException(String.format(EVENT_TYPE_NOT_FOUND_MSG, bundleName[0], appName[0], eventTypeName))
-                                                )
-                                                .onFailure().invoke(() -> {
-                                                    /*
-                                                     * A NoResultException was thrown because no EventType was found. The
-                                                     * message is therefore considered rejected.
-                                                     */
-                                                    rejectedCounter.increment();
-                                                })
-                                                .onItem().transformToUni(eventType -> {
+                                        return kafkaMessageDeduplicator.registerMessageId(messageId)
```

Comment:
I'm wondering if in the future we should write the outcome of the processing, like if it was a failure and at what point to retry the message.